tests/nn/moe/test_moe_layer.py

```diff
@@ -73,6 +73,56 @@ def test_forward(device):
     assert torch.allclose(input, output)
 
 
+# Test Gate which round-robin routes tokens to experts
+class RoundRobinGate(torch.nn.Module):
+    def __init__(self, model_dim, num_experts):
+        super().__init__()
+        self.model_dim = model_dim
+        self.num_experts = num_experts
+
+    def forward(self, input):
+        g, s, m = input.shape
+        capacity = 2 * s // self.num_experts
+        output = torch.zeros(g, s, self.num_experts, capacity, dtype=input.dtype, device=input.device)
```

Comment:
I wonder if you can do something like:

```
o = []
for I in range(self.num_experts):
  x = np.array((g, s, 1, capacity))
  x[:,:,:,self.num_experts*I:self.num_experts*(I+1)] = 1.0
  o.append(x)
o = np.stack(o)
output = torch.tensor(o)
```

I didn't write the exact np code above, but just rough idea. That would avoid explicitly looping over all indexes and the meaning is more readable too?